subsys/suit/platform/sdfw/src/runners/suit_run_nrf54h20.c

```diff
@@ -6,48 +6,115 @@
 
 #include <suit_cpu_run.h>
 #include <zephyr/logging/log.h>
-#include <suit_platform_internal.h>
 #include <drivers/nrfx_common.h>
+
+#ifdef CONFIG_SDFW_VPRS
 #include <sdfw/vprs.h>
+#endif /* CONFIG_SDFW_VPRS */
 
 #ifdef CONFIG_SDFW_RESET_HANDLING_ENABLED
 #include <sdfw/reset_mgr.h>
 #endif /* CONFIG_SDFW_RESET_HANDLING_ENABLED */
 
 LOG_MODULE_REGISTER(suit_plat_run, CONFIG_SUIT_LOG_LEVEL);
 
-int suit_plat_cpu_run(uint8_t cpu_id, intptr_t run_address)
+/** Bitmask indicating, which CPU IDs were invoked.
+ *
+ * @note Since it is not possible to invoke the same CPU twice or halt a CPU that
+ *       was not invoked, this value is used to detect those unsupported operations.
+ */
+static uint32_t running_cpus;
+
+suit_plat_err_t suit_plat_cpu_halt(uint8_t cpu_id)
 {
+	int ret = 0;
+
 	switch (cpu_id) {
 	case NRF_PROCESSOR_APPLICATION: /* AppCore */
-	case NRF_PROCESSOR_RADIOCORE: { /* RadioCore */
+	case NRF_PROCESSOR_RADIOCORE:	/* RadioCore */
 #ifdef CONFIG_SDFW_RESET_HANDLING_ENABLED
-		/* Single run address implies no NSVTOR, so keep at reset value of 0x0. */
-		return reset_mgr_init_and_boot_processor(cpu_id, run_address, 0);
-#else
-		return SUIT_SUCCESS;
+		if ((running_cpus & BIT(cpu_id)) != 0) {
+			LOG_INF("Halting Cortex core %d", cpu_id);
+			ret = reset_mgr_boot_processor(cpu_id, true);
```

Comment:
Shouldn't we currently return some "unsupprted" error instead?

Reply:
Changed to `SUIT_PLAT_ERR_UNSUPPORTED`